### PR TITLE
Refactor #9291: Move easter egg code to Guest

### PR DESCRIPTION
**Pull request by 41northstudios**
Refactor static Peep functions related to easter egg names into member
functions in the Guest struct.

### CODE REVIEW

File: src/openrct2/ride/Station.cpp [hunk 1]
```diff
@@ -284,23 +284,31 @@ static void ride_race_init_vehicle_speeds(Ride* ride)
         {
             Peep* peep = &get_sprite(vehicle->peep[0])->peep;
 
-            switch (peep_get_easteregg_name_id(peep))
+            // Easter egg names should only work on guests
+            // Check to make sure this peep is a guest before applying easter egg logic
+
+            Guest* guest = peep->AsGuest();
+
+            if (guest)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Station.cpp]: Always explicitly check for nullptr:
```suggestion
            if (guest != nullptr)
```

File: src/openrct2/ride/Station.cpp [hunk 2]
```diff
@@ -284,23 +284,31 @@ static void ride_race_init_vehicle_speeds(Ride* ride)
         {
             Peep* peep = &get_sprite(vehicle->peep[0])->peep;
 
-            switch (peep_get_easteregg_name_id(peep))
+            // Easter egg names should only work on guests
+            // Check to make sure this peep is a guest before applying easter egg logic
```
Comment (Gymnasiast) [inline src/openrct2/ride/Station.cpp]: This part of the comment is obvious from the code. Keep the bit on line 287, but ditch the one on 288.

File: src/openrct2/actions/GuestSetNameAction.hpp [hunk 3]
```diff
@@ -113,7 +113,15 @@ DEFINE_GAME_ACTION(GuestSetNameAction, GAME_COMMAND_SET_GUEST_NAME, GameActionRe
 
         peep_update_name_sort(peep);
 
-        peep_handle_easteregg_name(peep);
+        
+        // Easter egg functions are for guests only
+        // Make sure this is a guest before trying to handle an easter egg name
```
Comment (Gymnasiast) [inline src/openrct2/actions/GuestSetNameAction.hpp]: Ditch this line, it is obvious from the code. Keep the bit on line 117.

File: src/openrct2/peep/Peep.cpp [hunk 4]
```diff
@@ -1271,9 +1245,9 @@ void Peep::Update()
     if (carryCheck <= 255)
     {
         auto guest = AsGuest();
-        if (guest != nullptr)
+        if (guest)
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: This line _used_ to be correct. Please revert.

File: src/openrct2/peep/Peep.h [hunk 5]
```diff
@@ -718,6 +718,7 @@ struct Peep : rct_sprite_common
     void RemoveFromQueue();
     void RemoveFromRide();
     void InsertNewThought(PeepThoughtType thought_type, uint8_t thought_arguments);
+
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.h]: Unnecessary new line. Try to keep your diff as clean as possible.

File: src/openrct2/peep/Peep.h [hunk 6]
```diff
@@ -775,6 +776,12 @@ struct Guest : Peep
     bool DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price);
     void SetSpriteType(PeepSpriteType new_sprite_type);
 
+    void HandleEasterEggName();
+    int32_t GetEasterEggNameId();
+    void UpdateEasterEggInteractions();
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.h]: Three blank lines are two too many.
This comment applies to the whole PR - I won't spam this comment ;-)

**Comment by Gymnasiast** [comment]
Besides addressing my comment you'll need to rebase this PR. In general, make sure your branch is rebased against `develop` when you're opening a PR.

File: src/openrct2/peep/Guest.cpp [hunk 7]
```diff
@@ -369,6 +396,298 @@ static void peep_head_for_nearest_ride_type(Guest* peep, int32_t rideType);
...
+        }
+    }
+
+    if (peep_flags & PEEP_FLAGS_ICE_CREAM)
+    {
+        ApplyEasterEggToNearbyGuests(&Guest::GivePassingPeepsIceCream);
+    }
+}
+
+int32_t Guest::GetEasterEggNameId()
```
Comment (janisozaur) [inline src/openrct2/peep/Guest.cpp]: Const

**Pull request merged**